File: cocos/scripting/lua-bindings/auto/api/WebView.lua

--------------------------------
-- @module WebView
-- @extend Widget
-- @parent_module ccexp

--------------------------------
-- Gets whether this WebView has a back history item.<br>
-- return WebView has a back history item.
-- @function [parent=#WebView] canGoBack 
-- @param self
-- @return bool#bool ret (return value: bool)
        
--------------------------------
-- Sets the main page content and base URL.<br>
-- param string The content for the main page.<br>
-- param baseURL The base URL for the content.
-- @function [parent=#WebView] loadHTMLString 
-- @param self
-- @param #string string
-- @param #string baseURL
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- Goes forward in the history.
-- @function [parent=#WebView] goForward 
-- @param self
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- Goes back in the history.
-- @function [parent=#WebView] goBack 
-- @param self
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- Set WebView should support zooming. The default value is false.
-- @function [parent=#WebView] setScalesPageToFit 
-- @param self
-- @param #bool scalesPageToFit
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- Loads the given fileName.<br>
-- param fileName Content fileName.
-- @function [parent=#WebView] loadFile 
-- @param self
-- @param #string fileName
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- @overload self, string, bool         
-- @overload self, string         
-- @function [parent=#WebView] loadURL
-- @param self
-- @param #string url
-- @param #bool cleanCachedData
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)

--------------------------------
-- Set whether the webview bounces at end of scroll of WebView.
-- @function [parent=#WebView] setBounces 
-- @param self
-- @param #bool bounce
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- Evaluates JavaScript in the context of the currently displayed page.
-- @function [parent=#WebView] evaluateJS 
-- @param self
-- @param #string js
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- Get the Javascript callback.
-- @function [parent=#WebView] getOnJSCallback 
-- @param self
-- @return function#function ret (return value: function)
        
--------------------------------
-- Gets whether this WebView has a forward history item.<br>
-- return WebView has a forward history item.
-- @function [parent=#WebView] canGoForward 
-- @param self
-- @return bool#bool ret (return value: bool)
        
--------------------------------
-- Stops the current load.
-- @function [parent=#WebView] stopLoading 
-- @param self
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- Reloads the current URL.
-- @function [parent=#WebView] reload 
-- @param self
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- Set javascript interface scheme.<br>
-- see WebView::setOnJSCallback()
-- @function [parent=#WebView] setJavascriptInterfaceScheme 
-- @param self
-- @param #string scheme
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- Allocates and initializes a WebView.
-- @function [parent=#WebView] create 
-- @param self
-- @return experimental::ui::WebView#experimental::ui::WebView ret (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- 
-- @function [parent=#WebView] onEnter 
-- @param self
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- Toggle visibility of WebView.
-- @function [parent=#WebView] setVisible 
-- @param self
-- @param #bool visible
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- 
-- @function [parent=#WebView] onExit 
-- @param self
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
--------------------------------
-- Default constructor.
-- @function [parent=#WebView] WebView 
-- @param self
-- @return experimental::ui::WebView#experimental::ui::WebView self (return value: cc.experimental::ui::WebView)
        
return nil

File: tests/game-controller-test/proj.ohos/libSysCapabilities/src/main/ets/components/webview/WebView.ts
import { ViewRect, WebViewMsgEntity } from '../../entity/WorkerMsgEntity';

export class WebView {
    static MODULE_NAME: string = 'WebView';

    private static parentPort;

    static init(parentPort) : void {
        WebView.parentPort = parentPort;
    }

    static createWebView(viewTag: number) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'createWebView', viewTag);
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static removeWebView(viewTag: number) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'removeWebView', viewTag);
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static loadData(viewTag: number, data: string, mimeType: string, encoding: string, baseURL: string) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'loadData', viewTag);
        webViewMsgEntity.data = data;
        webViewMsgEntity.mimeType = mimeType;
        webViewMsgEntity.encoding = encoding;
        webViewMsgEntity.baseURL = baseURL;
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static loadURL(viewTag: number, url: string) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'loadURL', viewTag);
        webViewMsgEntity.url = url;
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static loadFile(viewTag: number, filePath: string) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'loadFile', viewTag);
        webViewMsgEntity.filePath = filePath;
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static stopLoading(viewTag: number) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'stopLoading', viewTag);
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static reload(viewTag: number) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'reload', viewTag);
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static canGoBack(viewTag: number) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'canGoBack', viewTag);
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static canGoForward(viewTag: number) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'canGoForward', viewTag);
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static goBack(viewTag: number) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'goBack', viewTag);
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static goForward(viewTag: number) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'goForward', viewTag);
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static setWebViewRect(viewTag: number, x: number, y: number, w: number, h: number) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'setWebViewRect', viewTag);
        let viewRect: ViewRect = new ViewRect(x, y, w, h);
        webViewMsgEntity.viewRect = viewRect;
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static setVisible(viewTag: number, visible: boolean) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'setVisible', viewTag);
        webViewMsgEntity.visible = visible;
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static setOpacityWebView(viewTag: number, opacity: number) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'setOpacityWebView', viewTag);
        webViewMsgEntity.opacity = opacity;
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static setBackgroundTransparent(viewTag: number) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'setBackgroundTransparent', viewTag);
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static evaluateJS(viewTag: number, js: string) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'evaluateJS', viewTag);
        webViewMsgEntity.js = js;
        WebView.parentPort.postMessage(webViewMsgEntity);
    }

    static setScalesPageToFit(viewTag: number, scalesPageToFit: boolean) : void {
        let webViewMsgEntity: WebViewMsgEntity = new WebViewMsgEntity(WebView.MODULE_NAME, 'setScalesPageToFit', viewTag);
        webViewMsgEntity.scalesPageToFit = scalesPageToFit;
        WebView.parentPort.postMessage(webViewMsgEntity);
    }
}